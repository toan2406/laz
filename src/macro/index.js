import { createMacro } from 'babel-plugin-macros';

module.exports = createMacro(({ references, babel: { types: t } }) => {
  if (references.thunk) {
    references.thunk.forEach(({ parentPath }) => {
      parentPath.replaceWith(t.numericLiteral(999));
    });
  }
});
