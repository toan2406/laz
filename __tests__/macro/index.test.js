import pluginTester from 'babel-plugin-tester';
import plugin from 'babel-plugin-macros';

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  tests: [
    `
      import { thunk } from '../../src/macro';

      thunk(1);
    `,
  ],
});
