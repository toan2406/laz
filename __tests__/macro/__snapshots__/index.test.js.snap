// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin 1. unknown plugin: 1. unknown plugin 1`] = `

import { thunk } from '../../src/macro';

thunk(1);

      ↓ ↓ ↓ ↓ ↓ ↓

() => 1;


`;
